crates/re_log_types/src/size_bytes_ext.rs

```diff
@@ -0,0 +1,90 @@
+    #[inline]
+    fn size_bytes(&self) -> u64 {
+        std::mem::size_of_val(self) as u64
+            + match self {
+                DataType::Null
+                | DataType::Binary
+                | DataType::Boolean
+                | DataType::Date32
+                | DataType::Date64
+                | DataType::Float16
+                | DataType::Float32
+                | DataType::Float64
+                | DataType::Int16
+                | DataType::Int32
+                | DataType::Int64
+                | DataType::Int8
+                | DataType::LargeBinary
+                | DataType::LargeUtf8
+                | DataType::UInt16
+                | DataType::UInt32
+                | DataType::UInt64
+                | DataType::UInt8
+                | DataType::Time32(_)
+                | DataType::Time64(_)
+                | DataType::Duration(_)
+                | DataType::Interval(_)
+                | DataType::FixedSizeBinary(_)
+                | DataType::Decimal(_, _)
+                | DataType::Decimal256(_, _)
+                | DataType::Utf8 => 0,
+                DataType::Timestamp(_, str) => str.as_ref().map_or(0, SizeBytesExt::size_bytes),
+                DataType::List(field)
+                | DataType::FixedSizeList(field, _)
+                | DataType::LargeList(field)
+                | DataType::Map(field, _) => field.size_bytes(),
+                DataType::Struct(fields) => fields.iter().map(SizeBytesExt::size_bytes).sum(),
+                DataType::Union(fields, indices, _) => {
+                    let indices: &Option<Vec<i32>> = indices; // safety
+                    fields.iter().map(SizeBytesExt::size_bytes).sum::<u64>()
+                        + indices.as_ref().map_or(0, |indices| {
+                            std::mem::size_of_val(indices.as_slice()) as u64
+                        })
+                }
+                DataType::Dictionary(_, datatype, _) => datatype.size_bytes(),
+                DataType::Extension(name, datatype, extra) => {
+                    datatype.size_bytes()
+                        + name.size_bytes()
+                        + extra.as_ref().map_or(0, SizeBytesExt::size_bytes)
+                }
+            }
+    }
+}
+
+impl SizeBytesExt for Field {
+    #[inline]
+    fn size_bytes(&self) -> u64 {
+        let Field {
+            name,
+            data_type,
+            is_nullable,
+            metadata,
+        } = self;
+
+        let is_nullable: &bool = is_nullable; // safety
+
+        data_type.size_bytes()
+            + name.size_bytes()
+            + std::mem::size_of_val(is_nullable) as u64
+            + metadata.keys().map(SizeBytesExt::size_bytes).sum::<u64>()
+            + metadata.values().map(SizeBytesExt::size_bytes).sum::<u64>()
+    }
+}
+
+impl SizeBytesExt for String {
+    #[inline]
+    fn size_bytes(&self) -> u64 {
+        (std::mem::size_of_val(self) + std::mem::size_of_val(self.as_bytes())) as u64
```

Comment:
Here you include `std::mem::size_of_val(self)`, but `DataType::size_bytes` does not

Reply:
it does?